doc/migrations/3.6.0_3.6.x.md

```diff
@@ -45,3 +45,39 @@ If you want to use the new password format with a full set of supported SHA hash
 Otherwise, please make sure that you provide the right `sasl_mechanism` configuration, where the mechanism you authenticate with is compatible with the password format you store.
 
 For more details related to the new password format, please refer to [authentication](../../Advanced-configuration#authentication) and [SCRAM serialization](../../scram-serialization.md) sections.
+
+## Message retraction
+
+Support for [XEP-0424: Message Retraction](http://xmpp.org/extensions/xep-0424.html) requires a new column for the `origin_id` attribute of MAM messages, which allows MAM to identify the messages to retract. Indexes for this column are required for efficient queries. Only the messages stored after this change can be retracted.
```

Comment:
Reading this sentence I can interpret it in a way that if I don't want the support for message retraction I don't have to add the column to the database. I think this assumption is not true, and we need the new column in order to archive the messages correctly regardless of the retraction functionality. Could you rephrase it in a way that this new column is mandatory?

Reply:
> A corner case just came to my mind, which as far as I can see we are handling in a good enough way.
> A user sends two different messages with the same `origin-id`, according to current schema they will be stored. When retracting, only the newer message will be retracted.
> @chrzaszcz do you think it'd be worth to document this behaviour and advise the client to use ids as unique as possible, like `UUIDs`?

Retracting only the most recent message is done on purpose. It is documented now, I also documented the way MIM finds the message to rectract.